l5kit/l5kit/evaluation/extract_losses.py

```diff
@@ -18,22 +18,29 @@ def extract_csv(csv_path: str) -> list:
     ground_truth_rows = extract_csv(ground_truth_path)[1:]
     inference_rows = extract_csv(inference_output_path)[1:]
 
+    assert len(ground_truth_rows[0]) == len(inference_rows[0]), "number of fields do not match"
+
+    future_num_frames = int(ground_truth_rows[0][2])  # read future_num_frames TODO: make this less mysterious
```

Comment:
Also not a huge fan of using these indices etc I have to say - maybe I'm not seeing it, but any reason why not work with column names (e.g. using np.savetxt for saving a csv or pandas should work)?

Reply:
Also connecting to e.g. the parse_values function in general, but maybe scope of a different PR.

Reply:
This will be completely re-written @oliver-scheel . It's a very old code indeed :)